thirdparty/autodiff/autodiff/forward/eigen.hpp

```diff
@@ -0,0 +1,344 @@
+//                  _  _
+//  _   _|_ _  _|o_|__|_
+// (_||_||_(_)(_|| |  |
+//
+// automatic differentiation made easier in C++
+// https://github.com/autodiff/autodiff
+//
+// Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+// Copyright (c) 2018-2020 Allan Leal
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+//------------------------------------------------------------------------------
+// SUPPORT FOR EIGEN MATRICES AND VECTORS OF DUAL
+//------------------------------------------------------------------------------
+namespace Eigen {
+
+template<typename T>
+struct NumTraits;
+
+template<typename T, typename G>
+struct NumTraits<autodiff::forward::Dual<T, G>> : NumTraits<autodiff::forward::ValueType<T>> // permits to get the epsilon, dummy_precision, lowest, highest functions
```

Comment:
Could you split this line?

Reply:
This is thirdparty code.

Reply:
I did not see that. Yes, of course, for thirdparty code it does not apply. This stays as-is.